discovery-provider/src/queries/query_helpers.py

```diff
@@ -751,6 +798,19 @@ def populate_playlist_metadata(session, playlist_ids, playlists, repost_types, s
     return playlists
 
 
+def get_playlist_repost_counts(session, playlist_ids, playlists, repost_types):
+    # build dict of playlist id --> repost count
+    playlist_repost_counts = dict(get_repost_counts(
```

Comment:
does this helper function not work in `get_track_repost_counts()`?

Reply:
Maybe a separate refactor. Worried about potentially subtle diffs between the two -- these were lifted directly out of the populate_x_metadata functions and probably should go through and clean those up there as well as add full unit tests

Reply:
makes sense 👍